cpu/stm32_common/periph/timer.c

```diff
@@ -53,8 +53,14 @@ int timer_init(tim_t tim, unsigned long freq, timer_cb_t cb, void *arg)
     dev(tim)->CR1  = 0;
     dev(tim)->CR2  = 0;
     dev(tim)->ARR  = timer_config[tim].max;
-    /* set prescaler */
+    /* set prescaler: the stm32f1 introduces a clock multiplier of 2 in the case
+     * the APB1 prescaler is != 1, so we need to catch that
+     * -> see reference manual section 7.2.1 */
+#if defined(CPU_FAM_STM32F1) && (CLOCK_APB1 != CLOCK_CORECLOCK)
```

Comment:
If I read the manual correctly
```C
if defined(CPU_FAM_STM32F1) && (CLOCK_APB1 < CLOCK_CORECLOCK)
```
would be more precise, but since `CLOCK_APB1` cannot be bigger than `CLOCK_CORECLOCK` anyway, that's doesn't change anything.